docs/src/capi/explanation/in-place-upgrades.md

```diff
@@ -0,0 +1,236 @@
+of tens, hundreds, or thousands of nodes, annotating them one by one and 
+keeping track of their statuses, failures, reasons and errors can become 
+a daunting or even impossible task. “Orchestrated In-Place Upgrade 
+Controller” to the rescue!
+
+The main idea behind this class of controllers is to enable us to upgrade 
+multiple machines by only annotating a single object: the owner of 
+those machines. In CAPI, we mostly have two main machine groups in 
+our workload cluster: “control-plane-node machines” and “worker-node 
+machines”.
+
+Let’s say we want to upgrade all of our worker nodes. In this case, we 
+only need to annotate the `MachineDeployment` object. The orchestrated 
+upgrade controller watches for the `upgrade-to` annotation on the 
+`MachineDeployment` and will trigger an in-place upgrade for all of 
+its owned machines by annotating them one by one. The responsibility 
+is then delegated to the single machine upgrader to make sure each 
+individual machine is getting upgraded successfully. The orchestrator 
+is only there to keep track of the upgrade status of these machines.
+If any of the machines fail to get upgraded, the orchestrator marks 
+the owner (here, `MachineDeployment`) with the respective annotations 
+(`status: failed` and `last-failed-attempt-at`), publishes helpful 
+and informative events and trusts the single machine controller to do 
+the retry and eventually succeed. When all the owned machines are 
+upgraded as expected, the orchestrator considers the operation successful, 
+marks the owner with `release` and `status: done` and steps down.
+
+To paint a better picture, let’s have a look the flow of the orchestrator:
+
+```{figure} ../../assets/in-place-upgrades/orchestrated.png
+   :width: 100%
+   :alt: Orchestrated In-Place Upgrade Flow
+
+   Orchestrated In-Place Upgrade Flow
+```
+
+For the detailed implementation of this controller, make sure to check out 
+the [{{product}} CAPI repository][capi-repo].
+
+#### Locking The Upgrade Process
+
+There might be scenarios where we need to make sure that only a limited 
+number of machines are going to get upgraded at the same time. An example 
+might be upgrading control plane machines. If multiple nodes become 
+unavailable due to getting upgraded, we might lose quorum, experience severe 
+downtime, or end up in an undesirable state that is very costly to get out of.
+
+Let’s say we only want to allow a single upgrade at any given point (no 
+parallel upgrades). A lock or a semaphore can be implemented to ensure 
+that the orchestrator is not going to trigger an upgrade for multiple 
+machines, even if multiple instances of the orchestrator are reconciling 
+the same object in parallel. Note that this implementation can be 
+generalized to allow `n` upgrades at the same time, instead of only 1.
+
+For the detailed implementation of the locking process, make sure to 
+checkout the [{{product}} CAPI repository][capi-repo].
+
```

Comment:
As a user all I care about is where do I configure `n`, please add that explanation and remove the details.

Reply:
Well, for now it can't be configured :)

Reply:
Alright, then let's remove this until we have a way to configure it.